packages/zcli-theme/src/lib/getRuntimeContext.test.ts

```diff
@@ -0,0 +1,123 @@
+import * as sinon from 'sinon'
+import * as fs from 'fs'
+import { expect, test } from '@oclif/test'
+import { CliUx } from '@oclif/core'
+import getRuntimeContext from './getRuntimeContext'
+
+const baseFlags = {
+  bind: 'localhost',
+  port: 1000,
+  logs: true,
+  host: 'localhost'
+}
+
+const credentialFlags = {
+  subdomain: 'z3n',
+  username: 'admin@zendesk.com',
+  password: '123456'
+}
+
+const flags = {
+  ...baseFlags,
+  ...credentialFlags
+}
+
+describe('getRuntimeContext', () => {
+  beforeEach(() => {
+    sinon.restore()
+  })
+
+  context('using CLI flags', () => {
+    it('uses the flags to generate the context', async () => {
+      const context = await getRuntimeContext('theme/path', flags)
+
+      expect(context).to.deep.equal({
+        bind: 'localhost',
+        host: 'localhost',
+        logs: true,
+        origin: 'https://z3n.zendesk.com',
+        port: 1000,
+        subdomain: 'z3n',
+        username: 'admin@zendesk.com',
+        password: '123456'
+      })
+    })
+  })
+
+  context('using a zcli.theme.config.json file', () => {
+    it('uses the config file to generate the context over the flags', async () => {
+      const existsSyncStub = sinon.stub(fs, 'existsSync')
+      const readFileSyncStub = sinon.stub(fs, 'readFileSync')
+
+      existsSyncStub
+        .withArgs('theme/path/zcli.theme.config.json')
+        .returns(true)
+
+      readFileSyncStub
+        .withArgs('theme/path/zcli.theme.config.json')
+        .returns(JSON.stringify({
+          subdomain: 'z3n',
+          username: 'admin@zendesk.com',
+          password: '123456'
+        }))
+
+      const context = await getRuntimeContext('theme/path', flags)
```

Comment:
I think we should pass `baseFlags` here, otherwise the credential are read from the flags instead of from the config file